lib/srv/reexec.go

```diff
@@ -612,9 +612,77 @@ func (o *osWrapper) startNewParker(ctx context.Context, credential *syscall.Cred
 	return nil
 }
 
-// RunForward reads in the command to run from the parent process (over a
+type forwardHandler func(ctx context.Context, addr string, file *os.File) error
+
+func handleLocalPortForward(ctx context.Context, addr string, file *os.File) error {
+	conn, err := uds.FromFile(file)
+	_ = file.Close()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	defer conn.Close()
+	var d net.Dialer
+	remote, err := d.DialContext(ctx, "tcp", addr)
+	if err != nil {
+		log.WithError(err).Errorf("error dialing %v", addr)
```

Comment:
Do you still want to call `ProxyConn` if there was an error dialing? I expect `remote` is likely nil in that case..